src/librawspeed/decoders/Cr2Decoder.cpp

```diff
@@ -131,11 +130,10 @@ RawImage Cr2Decoder::decodeNewFormat() {
   TiffEntry* offsets = raw->getEntry(STRIPOFFSETS);
   TiffEntry* counts = raw->getEntry(STRIPBYTECOUNTS);
 
-  LJpegPlain l(*mFile, offsets->getInt(), counts->getInt(), mRaw);
-  l.addSlices(s_width);
+  Cr2Decompressor d(*mFile, offsets->getInt(), counts->getInt(), mRaw);
```

Comment:
> If it is a function, how will it be able to cache memory allocations e.g.?

Not at all (without nasty `static` tricks of course). But it does not - that is what I tried to explicitly hint at with the `// executed exactly once` comment.

> i.e. they should only happen in the main thread.

Something like that will certainly require substantial changes. I think it is better to then introduce exactly what is necessary to achieve that than to now introduce something that is not used now but then might or might not be used in the current form later.